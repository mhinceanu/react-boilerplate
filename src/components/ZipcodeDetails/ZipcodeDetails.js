import React from 'react';
import PropTypes from 'prop-types';
import './ZipcodeDetails.scss';

const ZipcodeDetails = ({ details }) => {
  if (!details) {
    return <div data-testid="no-data">No data.</div>;
  }

  return (
    <article>
      <p>Country: {details?.country}</p>
      <p>Post code: {details?.['post code']}</p>

      {details?.places?.map((place) => (
        <section key={place.state}>
          <p>Latitude: {place?.latitude}</p>
          <p>Longitude: {place?.longitude}</p>
          <p>Place name: {place?.['place name']}</p>
          <p>State: {place?.state}</p>
          <p>Abbreviation: {place?.['state abbreviation']}</p>
        </section>
      ))}
    </article>
  );
};

ZipcodeDetails.propTypes = {
  details: PropTypes.objectOf(
    PropTypes.shape({
      country: PropTypes.string,
      ['post code']: PropTypes.string,
      places: PropTypes.arrayOf(
        PropTypes.shape({
          latitude: PropTypes.string,
          longitude: PropTypes.string,
          ['place name']: PropTypes.string,
          state: PropTypes.string,
          ['state abbreviation']: PropTypes.string
        })
      )
    })
  )
};

export default ZipcodeDetails;
