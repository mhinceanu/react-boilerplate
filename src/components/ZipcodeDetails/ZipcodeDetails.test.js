import React from 'react';
import { render } from '@testing-library/react';
import ZipcodeDetails from './ZipcodeDetails';

describe('<ZipcodeDetails />', function () {
  const getComponent = (data) => {
    return <ZipcodeDetails details={data} />;
  };

  it('should render <ZipcodeDetails /> with data', () => {
    const data = {
      'post code': '90210',
      country: 'United States',
      'country abbreviation': 'US',
      places: [
        {
          'place name': 'Beverly Hills',
          longitude: '-118.4065',
          state: 'California',
          'state abbreviation': 'CA',
          latitude: '34.0901'
        }
      ]
    };

    const { container } = render(getComponent(data));

    expect(container).toMatchSnapshot();
  });

  it('should render <ZipcodeDetails /> with no data', () => {
    const { container } = render(getComponent());

    expect(container).toMatchSnapshot();
  });
});
