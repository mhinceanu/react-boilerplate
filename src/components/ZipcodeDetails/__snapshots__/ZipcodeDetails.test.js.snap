// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ZipcodeDetails /> should render <ZipcodeDetails /> with data 1`] = `
<div>
  <article>
    <p>
      Country: 
      United States
    </p>
    <p>
      Post code: 
      90210
    </p>
    <section>
      <p>
        Latitude: 
        34.0901
      </p>
      <p>
        Longitude: 
        -118.4065
      </p>
      <p>
        Place name: 
        Beverly Hills
      </p>
      <p>
        State: 
        California
      </p>
      <p>
        Abbreviation: 
        CA
      </p>
    </section>
  </article>
</div>
`;

exports[`<ZipcodeDetails /> should render <ZipcodeDetails /> with no data 1`] = `
<div>
  <div
    data-testid="no-data"
  >
    No data.
  </div>
</div>
`;
